piptools/scripts/compile.py

```diff
@@ -33,6 +33,13 @@
 DEFAULT_REQUIREMENTS_OUTPUT_FILE = "requirements.txt"
 METADATA_FILENAMES = frozenset({"setup.py", "setup.cfg", "pyproject.toml"})
 
+# TODO: drop click 7 and remove this block, pass directly to version_option
+if click.__version__[0] == "7":
```

Comment:
This check is not very reliable: what if the version is 70?

Reply:
true, I wrongly expected this to be removed sooner, but better to be safe. What do you think about `click.__version__.split(".")[0] == "7"`? That looks uglier but would do what it should do.

Reply:
Almost. Except you should turn that into an integer and use `>` in the comparison. You may also want to move this into a compat module and declare some variable like `IS_CLICK_VER_8_PLUS`.

The reason for sooner vs later is not the best way of thinking about it is because over time the end users may use a release of pip-tools with this code which allows for any version of click above 7. So they may end up with old pinned pip-tools but a very modern click. I seem to recall stories of how win devs used a comparison with "9" for win 95/98 and then suddenly it was 2000, and then win 7/8 which started driving nuts old software that still works on new platforms because people never considered that versioning could change, back in the day.

Reply:
Definitely makes sense, thanks for the words!

implemented this, hope it is the proper place and implementation :)